serde/src/main/java/com/amazon/ionhiveserde/objectinspectors/IonStructToMapObjectInspector.java

```diff
@@ -80,12 +80,20 @@ public Object getMapValueElement(final Object data, final Object key) {
 
         final Map<String, IonValue> map = new HashMap<>(size, DEFAULT_LOAD_FACTOR);
         for (IonValue v : struct) {
-            map.put(v.getFieldName(), v);
+            if (!IonUtil.isIonNull(v)) {
+                map.put(v.getFieldName(), v);
+            }
```

Comment:
Is this right? In `{ foo: null }`, the Ion null has a field name. Is `null` not a valid value in a map?

Reply:
Discussed offline.